### ISSUE TITLE
PropertyFieldCollectionData onChanged missing

### ISSUE THREAD

**Issue by IRRDC** [pnp/sp-dev-fx-property-controls#297]
#### Category
- [x] Enhancement
- [ ] Bug
- [ ] Question

#### Version

Please specify what version of the library you are using: [1.20.0]

#### Expected / Desired Behavior / Question
WIthout an onChanged or onItemDeleted callback there is no way to perform data clean up in case a data row is deleted by the user.
There already is an onChangeCallback in the render method that is not part of the interface.

Thanks!

**Comment by AJIXuMuK** [comment]
Hi @IRRDC 
Could you please provide a bit more details of your scenario?

When you use `PropertyFieldCollectionData` you're providing `targetProperty` value.
The control will update the value of that property if something got changed.

```typescript
PropertyFieldCollectionData("collectionData", {
  key: "collectionData",
  // ...
  value: this.properties.collectionData,
  // ...
})
```

**Comment by IRRDC** [comment]
@AJIXuMuK I have switched to using the generel property changed event as a workaround. I was doing some interaction while the dialog was open (displaying all the entered values from all rows in one column as a hint) and was using the changed events from the controls in the rows but what I could not detect was when a row in the dialog got deleted (and I had to remove a value from my hint). Now I only update the hints after the dialog has been closed and the values have been saved, which is okay for now.
